### PR TITLE
Add changes for using gcc7.3 on AL2 with 4.14.94-89.73.amzn2.x86_64

### PR DESCRIPTION
**Pull request by awshkulkar**
This change only includes changes needed to build (compile/link) the XRT on AL2. A newPR will be submitted for creating install-able rpm package.

### CODE REVIEW

**Comment by maxzhen** [comment]
ok to test

**Comment by awshkulkar** [comment]
I will add the warning suppression conditionally for gcc version 7.3 and up. Pleas do not merge this commit yet.

File: src/runtime_src/xocl/CMakeLists.txt [hunk 1]
```diff
@@ -13,6 +13,8 @@ include_directories(
   ${XRT_XOCL_API_DIR}
 )
 
+add_compile_options("-Wno-nonnull")
```
Comment (stsoe) [inline src/runtime_src/xocl/CMakeLists.txt]: What's the reason for disabling this warning?

Comment (awshkulkar) [reply src/runtime_src/xocl/CMakeLists.txt]: The code is in boot devel package, not in XRT.

Comment (stsoe) [reply src/runtime_src/xocl/CMakeLists.txt]: I don't understand what that means?  This is a CMakeList.txt in XRT.

Comment (awshkulkar) [reply src/runtime_src/xocl/CMakeLists.txt]: I take it back. I was looking at a different log. You are right. I see the issue here:
 499   | if (flags & CL_MEM_COPY_HOST_PTR)
  500   | | std::memcpy(m_host_ptr,host_ptr,sz);
There is no null check. I will fix this code and then remove this supression. Will push a new commit.

**Pull request closed without merging**